### PR TITLE
Add list-sets command

### LINKED ISSUE #5621: Provide a command-line argument to list all existing rulesets
In PhpStorm there is a chooser to run a predefined ruleset but available rulesets are hard-coded since there is no easy way to fetch them from CS-Fixer. 
For now, we have the following standards: `"PSR1", "PSR2", "Symfony", "DoctrineAnnotation", "PHP70Migration", "PHP71Migration"` and looking into https://github.com/FriendsOfPHP/PHP-CS-Fixer/tree/2.18/src/RuleSet/Sets it looks like the list is far from complete.

Since we support different versions of PHP-CS-Fixer, a command-line option that outputs the available standards will allow us to list all rulesets instead of a limited list supported by 2.8.0 (the earliest version we support).

### PR DESCRIPTION
**Pull request by keradus**
closes #5621 
example: `php php-cs-fixer list-sets --format json`

### CODE REVIEW

File: doc/schemas/list-sets/schema.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (keradus) [inline doc/schemas/list-sets/schema.json:1]: @MaXal , would that fit your needs?
// part one: schema

File: tests/Console/Report/ListSetsReport/JsonReporterTest.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        static::assertJsonSchema($input);
+        static::assertJsonStringEqualsJsonString($expected, $input);
+    }
+
+    /**
+     * @return string
+     */
+    protected function createSimpleReport()
+    {
+        return '{
```
Comment (keradus) [inline tests/Console/Report/ListSetsReport/JsonReporterTest.php:47]: @MaXal , would that fit your needs?
// part two: example json result

File: doc/schemas/list-sets/schema.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "type": "object",
+    "properties": {
+        "sets": {
```
Comment (keradus) [inline doc/schemas/list-sets/schema.json:5]: I was wondering about this special nesting level (let me call it "envelope") -> for now, we don't really need it.
Yet, we had cases in past, when Rule was expecting configuration like `[foo, bar]` and we couldn't extend it, causing a BC break to introduce `{items => [foo, bar], otherOption: true}`. let's be future ready

File: tests/Console/Command/ListSetsCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    public function testListWithJsonFormat()
+    {
+        $commandTester = $this->doTestExecute([
+            '--format' => 'json',
+        ]);
+
+        $resultText = $commandTester->getDisplay();
+
+        static::assertJson($resultText);
```
Comment (kubawerlos) [inline tests/Console/Command/ListSetsCommandTest.php:48]: Do we check anywhere that this JSON matches schema? Maybe we could do this using https://github.com/estahn/phpunit-json-assertions.

File: tests/Console/Command/ListSetsCommandTest.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        static::assertSame(0, $commandTester->getStatusCode());
+    }
+
+    public function testListWithJsonFormat()
+    {
+        $commandTester = $this->doTestExecute([
+            '--format' => 'json',
+        ]);
+
+        $resultText = $commandTester->getDisplay();
```
Comment (kubawerlos) [inline tests/Console/Command/ListSetsCommandTest.php:46]:
```suggestion
        $resultJson = $commandTester->getDisplay();
```

**Comment by kubawerlos** [comment]
Why did the coverage dropped?

File: tests/Console/Command/ListSetsCommandTest.php (hunk 4 above)
Comment (keradus) [reply tests/Console/Command/ListSetsCommandTest.php:48]: `ListSetsCommandTest` is only checking if the command does not crash here.
the check of the content and schema is executed in tests for reporters, eg here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5635/files#diff-153de19a2bcd8d41652d1edadfdc0df1f971ee09364e219e89a9b2187a0d1addR66

File: tests/Console/Command/ListSetsCommandTest.php (hunk 5 above)
Comment (keradus) [reply tests/Console/Command/ListSetsCommandTest.php]: :+1:

**Comment by keradus** [comment]
no clue

**Review by kubawerlos — APPROVED** [review]

**Comment by MaXal** [comment]
Thank you a lot for your work! The result works great! I choose to implement a parser based on `txt` format since it's much easier to parse.
image

**Comment by keradus** [comment]
Thanks for the feedback, @MaXal !
In that case, I gonna merge the PR as-is.

Please, be warned that for JSON, the result is well-specified and you can rely on [schema](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/515fb741ef6f5b454e990a7421d964bdb5ce7cc0/doc/schemas/list-sets/schema.json) under BC promise. For txt output, the format of message can be modified at any PHP CS Fixer update.

**Comment by MaXal** [comment]
@keradus Thank you! I will take a look at JSON format then to make the support more bullet-proof!

**Pull request merged**